ui/src/app/pages/admin/user/admin-user-profile.tsx

```diff
@@ -754,16 +757,31 @@ export const AdminUserProfile = (spinnerProps: WithSpinnerOverlayProps) => {
             <h2>Egress event history</h2>
           </FlexRow>
           <FlexRow>
-            <EgressEventsTable
-              displayPageSize={10}
-              sourceUserEmail={updatedProfile.username}
-            />
+            {renderIfAuthorized(
```

Comment:
Here are the main changes: don't render the component if your user doesn't have the access to call its endpoint.  Show some "no access" text instead, in that case